manager/controllers/app/m4dapplication_controller.go

```diff
@@ -290,16 +289,29 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 // ConstructIngestDataInfo already has the information about the external data set's endpoint,
 // but it adds the governance decisions. It also already has the link to the credentials, which was provided by the user.
 // NOTE: Assumes the credentials were put into the M4D credential manager by the user!!
-func (r *M4DApplicationReconciler) ConstructIngestDataInfo(externalDataset app.ExternalDataContext, req *modules.DataInfo, input *app.M4DApplication, clusters []m.Cluster) error {
+func (r *M4DApplicationReconciler) ConstructIngestDataInfo(externalDataset app.ExternalDataContext, req *modules.DataInfo, input *app.M4DApplication, clusters []multicluster.Cluster) error {
 	dataDetails := pb.DatasetDetails{
 		DataOwner:  "",
 		DataStore:  externalDataset.ExternalStore.Connection,
 		DataFormat: externalDataset.ExternalStore.Format,
 		Geo:        string(externalDataset.Residency),
 	}
+	// Store link to credentials rather than the credentials themselves
+	// This is a hack, but works for now.
+	creds := pb.Credentials{
```

Comment:
Please see https://github.com/IBM/the-mesh-for-data/pull/251. It keeps a path rather than the credentials themselves. Removes the need in this work-around.

Reply:
Agreed, but it only solves part of the problem. There is no API to get the credential path from the inline catalog.

Reply:
Code now takes the credentials from the inline catalog and copies them into vault